app/models/name/lifeform.rb

```diff
@@ -47,33 +47,91 @@ def validate_lifeform
 
   # Add lifeform (one word only) to all children.
   def propagate_add_lifeform(lifeform)
-    concat_str = Name.connection.quote("#{lifeform} ")
-    search_str = Name.connection.quote("% #{lifeform} %")
-    Name.connection.execute(%(
-      UPDATE names SET lifeform = CONCAT(lifeform, #{concat_str})
-      WHERE id IN (#{all_children.map(&:id).join(",")})
-        AND lifeform NOT LIKE #{search_str}
-    ))
-    Name.connection.execute(%(
-      UPDATE observations SET lifeform = CONCAT(lifeform, #{concat_str})
-      WHERE name_id IN (#{all_children.map(&:id).join(",")})
-        AND lifeform NOT LIKE #{search_str}
-    ))
+    concat_str = Name.connection.quote_string("#{lifeform} ")
+    search_str = Name.connection.quote_string("% #{lifeform} %")
+
+    for type in %w[name observation] do
+      update_manager = arel_update_add_lifeform(
+        type, concat_str, search_str
+      )
+      # puts(all_children.map(&:id).inspect)
+      # puts(update_manager.to_sql)
+      Name.connection.update(update_manager.to_sql)
+    end
```

Comment:
Yikes, all_children is potentially very expensive and not cached!  Also, can we use AR's update_all instead.  I don't see a way out of using CONCAT, though.

`name_ids = all_children.map(&:id)`
`Name.where(id: name_ids).and.not(Name[:lifeform].matches("% #{lifeform} %")).`
`           update_all(lifeform: Name[:lifeform] + "#{lifeform} ")`
`Observation.where(name_id: name_ids).and.not(Observation[:lifeform].matches("% #{lifeform} %")).`
`           update_all(lifeform: Observation[:lifeform] + "#{lifeform} ")`

Did I shorten "Arel::Nodes::SqlLiteral.new((table[:column] + string).to_sql)" too much?  Does it need the .to_sql?

`...update_all(lifeform: (Observation[:lifeform] + "#{lifeform} ").to_sql)`

Since update_all can take SQL, I assume Arel::Nodes::SqlLiteral must be unnecessary.